core/cas-server-core-events/src/test/java/org/apereo/cas/support/events/CasAuthenticationEventListenerTests.java

```diff
@@ -135,20 +160,94 @@ public void verifyCasRiskyAuthenticationDetectedEvent() {
             CoreAuthenticationTestUtils.getAuthentication(),
             CoreAuthenticationTestUtils.getRegisteredService(),
             new Object());
+        var current = numEventsReceived.get();
         applicationContext.publishEvent(event);
+        sleep(current + 1);
         assertFalse(casEventRepository.load().findAny().isEmpty());
     }
 
     @Test
     public void verifyCasTicketGrantingTicketDestroyed() {
         val event = new CasTicketGrantingTicketDestroyedEvent(this,
             new MockTicketGrantingTicket("casuser"));
+        var current = numEventsReceived.get();
         applicationContext.publishEvent(event);
+        sleep(current + 1);
         assertFalse(casEventRepository.load().findAny().isEmpty());
     }
+    @Test
+    public void verifyEventRepositoryHasOneEventOnly() {
+        clearEventRepository();
+        val event = new CasTicketGrantingTicketDestroyedEvent(this,
+                new MockTicketGrantingTicket("casuser"));
+        var current = numEventsReceived.get();
+        applicationContext.publishEvent(event);
+        sleep(current + 1);
```

Comment:
These 3 lines seems to have been repeated in almost all tests. Opportunity to refactor.

Reply:
see publishEventAndWaitToProcess